### ISSUE TITLE
Error while checking for engine updates

### ISSUE THREAD

**Issue by panther2** [triplea-game/triplea#4196]
### Engine version
1.9.0.0.12461

### My Operating System
Windows 7 64bit

### Can you describe how to trigger the error? (eg: what sequence of actions will recreate it?)

Start TripleA, see:

![traerror](https://user-images.githubusercontent.com/20812775/47252892-fef67300-d44b-11e8-853b-a42bb9c7e278.PNG)

### Do you have the exact error text? Please copy/paste if so

```
Okt 20, 2018 9:37:59 AM games.strategy.engine.auto.update.EngineVersionCheck checkForLatestEngineVersionOut
SCHWERWIEGEND: Error while checking for engine updates
java.util.NoSuchElementException: No value present
	at java.util.Optional.get(Unknown Source)
	at games.strategy.triplea.settings.GameSetting.value(GameSetting.java:47)
	at games.strategy.engine.auto.update.EngineVersionCheck.isEngineUpdateCheckRequired(EngineVersionCheck.java:59)
	at games.strategy.engine.auto.update.EngineVersionCheck.isEngineUpdateCheckRequired(EngineVersionCheck.java:44)
	at games.strategy.engine.auto.update.EngineVersionCheck.checkForLatestEngineVersionOut(EngineVersionCheck.java:25)
	at games.strategy.engine.auto.update.UpdateChecks.checkForUpdates(UpdateChecks.java:26)
	at java.lang.Thread.run(Unknown Source)

Okt 20, 2018 9:37:59 AM games.strategy.engine.framework.GameRunner lambda$start$0
SCHWERWIEGEND: No value present
java.util.NoSuchElementException: No value present
	at java.util.Optional.get(Unknown Source)
	at games.strategy.triplea.settings.GameSetting.value(GameSetting.java:47)
	at games.strategy.engine.auto.update.UpdatedMapsCheck.isMapUpdateCheckRequired(UpdatedMapsCheck.java:28)
	at games.strategy.engine.auto.update.UpdatedMapsCheck.isMapUpdateCheckRequired(UpdatedMapsCheck.java:16)
	at games.strategy.engine.auto.update.UpdateChecks.checkForUpdates(UpdateChecks.java:28)
	at java.lang.Thread.run(Unknown Source)
```

### Any additional information that may help
I have never experienced this before.
Not sure whether this might be relevant: I have updated Java 8 from update 181 to 191 before.
(Clean install).

I have never been aware of an automatic engine update check before. Maybe because I update manually on a daily base. Is there an option to disable that?

**Comment by ssoloff** [comment]
> I have never been aware of an automatic engine update check before. Maybe because I update manually on a daily base.

This check runs every two days.  However, it only prompts you to upgrade if one of the non-build-ID components of the version have changed compared to what's in the file https://raw.githubusercontent.com/triplea-game/triplea/master/latest_version.properties.  For example, if you were running 1.8.0.0, you would get prompted to upgrade because the latest version in the above file is 1.9.0.0.

It does _not_ prompt you to upgrade simply when the build ID changes (e.g. 1.9.0.0.9687 -> 1.9.0.0.12226).

> Is there an option to disable that?

There doesn't appear to be any way to turn it off.  Sounds like a good feature request. :smile:

**Comment by ssoloff** [comment]
@panther2 Should be fixed in 1.9.0.0.12469 or later.  I fixed a similar issue in the map update check, but that only runs once a month, so hopefully you'll get a new pre-release installed before you hit it. :smile:

**Comment by panther2** [comment]
Thank you, @ssoloff  :smile:
